lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift

```diff
@@ -0,0 +1,312 @@
+import SwiftSyntax
+import SwiftSyntaxMacros
+import SwiftDiagnostics
+
+public struct DebugDescriptionMacro {}
+
+/// Attached peer macro which converts `debugDescription` (or `description`) implementations into
+/// an LLDB type summary. The type summary record is emitted into a custom section (via a global
+/// constant), where LLDB will load from at debug time.
+extension DebugDescriptionMacro: PeerMacro {
+    public static func expansion<Decl, Context>(
+        of node: AttributeSyntax,
+        providingPeersOf declaration: Decl,
```

Comment:
ironically, I wrote it this way because of a bug in lldb where generics work, but the `some T` equivalent doesn't. So this use is only to aid debugging.